lib/compress/zstd_compress.c

```diff
@@ -2131,14 +2137,31 @@ static size_t ZSTD_resetCCtx_byCopyingCDict(ZSTD_CCtx* cctx,
                                                             : 0;
         size_t const hSize =  (size_t)1 << cdict_cParams->hashLog;
 
-        ZSTD_memcpy(cctx->blockState.matchState.hashTable,
-               cdict->matchState.hashTable,
-               hSize * sizeof(U32));
+        if (ZSTD_CDictIndicesAreTagged(&cctx->appliedParams.cParams)){
+            size_t i;
+            for (i = 0; i < hSize; i++) {
+                U32 const taggedIndex = cdict->matchState.hashTable[i];
+                cctx->blockState.matchState.hashTable[i] = taggedIndex >> ZSTD_SHORT_CACHE_TAG_BITS;
```

Comment:
Nit: I have a slight general preference towards decomposing the arithmetic and the memory operation. E.g., this would maybe better be:

```
U32 const taggedIndex = cdict->matchState.hashTable[i];
U32 const index = taggedIndex >> ZSTD_SHORT_CACHE_TAG_BITS;
cctx->blockState.matchState.hashTable[i] = index;
```